pkg/tsdb/cloudwatch/routes/dimension_keys_test.go

```diff
@@ -0,0 +1,80 @@
+package routes
+
+import (
+	"fmt"
+	"net/http"
+	"net/http/httptest"
+	"testing"
+
+	"github.com/grafana/grafana-plugin-sdk-go/backend"
+	"github.com/grafana/grafana/pkg/tsdb/cloudwatch/mocks"
+	"github.com/grafana/grafana/pkg/tsdb/cloudwatch/models"
+	"github.com/stretchr/testify/assert"
+)
+
+func Test_DimensionKeys_Route(t *testing.T) {
+	origNewListMetricsService := newListMetricsService
+	t.Cleanup(func() {
+		newListMetricsService = origNewListMetricsService
+	})
+	var clientFactoryMock = func(pluginCtx backend.PluginContext, region string) (clients models.ClientsProvider, err error) {
+		return nil, nil
+	}
+
```

Comment:
None of these are used.
In the tests below the clientFactoryMock is not used, can be replaced with `nil` which clearly communicates that it is not met in the code path.